### PR TITLE
Bootable Kernel

### PR DESCRIPTION
**Pull request by JT2M0L3Y**
## Changes
* can boot to a QEMU virtual environment using the resulting binary
  * simply prints "Hello World!" for now, no input possible

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -1,23 +1,7 @@
 [package]
+edition = "2021"
```
Comment (kevinthedang) [inline Cargo.toml:2]: Is there a reason why the edition is **2021**?

File: src/main.rs [hunk 2]
```diff
@@ -3,15 +3,42 @@
...
+    for (i, &byte) in HELLO.iter().enumerate() {
+        // unsafe used b/c of modifying bytes directly
+        unsafe {
+            *vga_buffer.offset(i as isize * 2) = byte;
+            *vga_buffer.offset(i as isize * 2 + 1) = TEXT_COLOR;
+        }
+    }
+
+    loop {}
+}
```
Comment (kevinthedang) [inline src/main.rs:44]: I assume this is the following stages:

1. Startup
2. Loop Forever after startup

Maybe I am wrong but it does start with `_start`

File: Cargo.toml (hunk 1 above)
Comment (JT2M0L3Y) [reply Cargo.toml:2]: The `edition` field marks the latest set of changes to Rust that a project is compiled with. This is separate from what version of the Rust language is used.

File: src/main.rs (hunk 2 above)
Comment (JT2M0L3Y) [reply src/main.rs:44]: Yes, think of `_start()` as the `main()` function. It loops forever after printing "Hello World".

**Review by kevinthedang — APPROVED** [review]

**Pull request merged**